src/control/security/config.go

```diff
@@ -105,6 +108,30 @@ type TransportConfig struct {
 	CertificateConfig `yaml:",inline"`
 }
 
+// TelemetryConfig contains all the information on whether or not to use
+// secure endpoint for telemetry and their location if their use is specified.
+type TelemetryConfig struct {
+	Port          int           `yaml:"port,omitempty"`
+	AllowInsecure bool          `yaml:"allow_insecure"`
+	Enabled       bool          `yaml:"enabled,omitempty"`
```

Comment:
Is `enabled` strictly necessary as a yaml option? IMO if a port is defined, we are using telemetry.